### PR TITLE
Add initial support for multi-qubit gates.

### PR DESCRIPTION
**Pull request by sergeisakov**
This PR also moves applying measurement gates from run_qsim.h to gate_appl.h.

### CODE REVIEW

File: lib/gate_appl.h [hunk 1]
```diff
@@ -62,50 +63,150 @@ inline void CalcMatrix4(unsigned q0, unsigned q1,
...
+    }
+  } else {
+    ApplyGate(simulator, gate, state);
   }
+
+  return true;
+}
+
+/**
+ * Applies the given gate to the simulator state.
```
Comment (95-martin-orion) [inline lib/gate_appl.h:115]: This comment should explicitly note that the function discards measurement results.

File: lib/gate_appl.h [hunk 2]
```diff
@@ -62,50 +63,150 @@ inline void CalcMatrix4(unsigned q0, unsigned q1,
...
+    }
+  } else {
+    ApplyFusedGate(simulator, gate, state);
+  }
+
+  return true;
+}
+
+/**
+ * Applies the given fused gate to the simulator state.
```
Comment (95-martin-orion) [inline lib/gate_appl.h:193]: This comment should explicitly note that the function discards measurement results.

**Review by 95-martin-orion — APPROVED** [review]

File: lib/gate_appl.h (hunk 1 above)
Comment (sergeisakov) [reply lib/gate_appl.h]: Done.

File: lib/gate_appl.h (hunk 2 above)
Comment (sergeisakov) [reply lib/gate_appl.h]: Done.

**Pull request merged**